syntax = "proto3";

package faceRecognition;

//This tells gRPC we have an InferenceServer service with an inference function, notice that we need to specify the type of the messages: InferenceRequest and InferenceReply

// "repeated" means list of


service FaceRecognitionService {
    rpc Inference (FaceRecognitionRequest) returns (FaceRecognitionInferenceReply) { }
}


//Request message containing image
message FaceRecognitionRequest {
    bytes image = 1;
    map<string, int32> personBox = 2;

}

//Response Message 
message FaceRecognitionInferenceReply {
    optional string uuid = 1;
    optional Box box = 2;

}

message Box {
    uint32 x1 = 1;
    uint32 y1 = 2;
    uint32 x2 = 3;
    uint32 y2 = 4;

}

